src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,48 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.UserNotAllowed;
 import core.basesyntax.model.User;
+import java.util.HashSet;
+import java.util.Set;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_LOGIN_LENGTH = 6;
+    private static final int MINIMAL_USER_AGE = 18;
+    private static final int MINIMAL_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
+    private Set<String> loginSet = new HashSet<>();
+
+    private void loginSetInit() {
+        for (User user : Storage.people) {
+            loginSet.add(user.getLogin());
+        }
+    }
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws UserNotAllowed {
+        if (user != null) {
+            loginSetInit();
```

Comment:
Initializing the `loginSet` inside the `register` method is not efficient. This should be done once, possibly in the constructor, to avoid re-initializing the set on every registration attempt.